### PR TITLE
Add WebP support to imglab

### PR DESCRIPTION
**Pull request by arrufat**
It should have been included in #2565.
I just noticed imglab couldn't find a dataset with WebP images.

### CODE REVIEW

File: tools/imglab/src/main.cpp [hunk 1]
```diff
@@ -552,9 +564,19 @@ int tile_dataset(const command_line_parser& parser)
...
     if (ext == "png")
+    {
         save_png(tile_images(images), out_image);
+    }
+    else if (ext == "jpg")
+    {
+        save_jpeg(tile_images(images), out_image, JPEG_QUALITY);
+    }
     else
-        save_jpeg(tile_images(images), out_image);
```
Comment (davisking) [inline tools/imglab/src/main.cpp:557]: jpeg is a better default for this.  It's really small and more importantly dlib will pretty much always end up being built with jpeg support, but not webp support.

Comment (arrufat) [reply tools/imglab/src/main.cpp:557]: I thought about it, but to reach this point, the extension must be either ".png", ".jpg", or ".webp", any other extension will cause the program to exit at line 513, (525 after this PR). So, I think it's fine, right? We will always use the extension requested by the user.

Comment (arrufat) [reply tools/imglab/src/main.cpp:557]: But maybe I can change the order of the if/else statements if that makes it more clear :)

Comment (davisking) [reply tools/imglab/src/main.cpp:557]: Oh right, yeah duh.  This was fine :D

**Comment by davisking** [comment]
Sweet, thanks for the PR :)

**Comment by arrufat** [comment]
I see that CI on Travis for windows is failing due to not having WebP support enabled. But not on GitHub Actions, though.
I think we're not building the `tools` in GitHub (imglab, htmlify, etc).

I think we have 2 options:
- either add some #ifdef guards around the WebP code (maybe not ideal, but it's nice to have WebP support)
- remove support for writing WebP from imglab, and add only support for reading by modifying the `match_endings`, which is probably what most end users will care about: finding WebP images if they exist in their dataset and therefore, they should bother enabling WebP support.

What do you think?

**Pull request merged**